modules/ROOT/pages/websocket/websocket-connector-config-topics.adoc

```diff
@@ -0,0 +1,285 @@
+</websocket:config>
+----
+
+Now we can reference this configuration from both inbound and outbound sockets.
+
+[source,xml,linenums]
+----
+<flow="acceptIncomingMessagesFlow">
+    <websocket:inbound-listener path="/quotes" config-ref="ws" />
+    <logger info="INFO" message="I listen for incoming messages at '/quotes'!"/>
+</flow>
+
+<flow="connectToWebSocketFlow">
+    <websocket:open-outbound-socket path="/chat" config-ref="ws"/>
+    <logger 
+      info="INFO" 
+      message="Open a connection to an external service listening at path '/chat'!"/>
+</flow>
+----
+
+=== TLS Configuration
+
+TLS (Transport Layer Security) allows for client and server apps to communicate through a secure channel and prevent attackers from eavesdropping or tampering with the transmitted data.
+
+In the case of the WebSocket server, TLS configuration is inherited from the referenced HTTP listener configuration, for example, to configure TLS on a WebSocket listener, point to `<http:listener-config>` with TLS configured.
+
+You first need to set up your trust store if your application connects to external services, a key store if your application listens for incoming connections, or you can set up both.
+
+In the following example, the WebSocket server inherits its TLS configuration from the HTTP listener, which in turn references  `<tls:context>` to configure a key store:
+
+[source,xml,linenums]
+----
+<tls:context name="listenerTlsContext" >
+    <tls:key-store 
+      path="tls/muleKeystore" 
+      keyPassword="mulepassword" 
+      password="mulepassword" 
+      alias="muleserver" />
+</tls:context>
+
+<http:listener-config name="listenerTlsConfig">
+    <http:listener-connection 
+      protocol="HTTPS" 
+      host="localhost" 
+      port="${listenerPort}" 
+      tlsContext="listenerTlsContext"/>
+</http:listener-config>
+
+<websocket:config name="requestConfigWithCertificate">
+    <websocket:connection>
+        <websocket:server-settings 
+         listenerConfig="listenerTlsConfig" 
+         listenerBasePath="/"/>
+    </websocket:connection>
+</websocket:config>
+----
+
+In case of a WebSocket client, you can set up a trust store by directly referencing the <tls:context> in the `tlsConfig` property.
+
+[source,xml,linenums]
+----
+<tls:context name="requestTlsContextWithCertificate" >
+    <tls:trust-store path="tls/trustStore" password="mulepassword" />
+</tls:context>
+
+<websocket:config name="requestConfigWithCertificate">
+    <websocket:connection>
+        <websocket:client-settings 
+         host="localhost" 
+         port="${listenerPort}" 
+         tlsContext="requestTlsContextWithCertificate" 
+         protocol="WSS"/>
+    </websocket:connection>
+</websocket:config>
+----
+
+The protocol parameter is optional and defaults to WS (insecure). If set to `WSS`, then SSL is enabled in the connection. If  `<tls:content>` was specified, then that configuration is used to create the SSL session. Otherwise, default settings are used.
```

Comment:
```suggestion
The `protocol` parameter is optional and defaults to WS (insecure). If set to `WSS`, then SSL is enabled in the connection. If `<tls:content>` was specified, then that configuration is used to create the SSL session. Otherwise, default settings are used.
```